src/modelHandler_OpenCL.cpp

```diff
@@ -313,7 +316,8 @@ namespace w2xc
 		const char *dev_name = proc->dev_name;
 		bool bin_avaiable = false;
 
-#if ((defined __linux) && !(defined __ANDROID__)) || _WIN32
+#if ((defined __linux) && !(defined __ANDROID__)) || defined(_WIN32) || \
+	defined(__DragonFly__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__)
```

Comment:
Is there no cleaner way to check for *BSD variants than a million defined checks ?

Reply:
`defined(BSD)` but it's only defined if `<sys/param.h>` is included, haven't been updated since BSD 4.4 Lite2 and is also defined on macOS. sysctl code is kernel-specific e.g., `KERN_PROC_PATHNAME` isn't supported on OpenBSD (for security reasons) or macOS (uses `_NSGetExecutablePath`); while `/proc` has been removed on DragonFly + OpenBSD and deprecated on FreeBSD + NetBSD.

Does current version look better?

Reply:
Also, `GENERATE_BINARY` can probably be enabled on all platforms. If getting absolute path to executable isn't supported/implemented current directory can be used as a fallback.